sycl/doc/extensions/proposed/sycl_ext_oneapi_fpga_mem.asciidoc

```diff
@@ -0,0 +1,507 @@
+    template <bool Enable>
+    using value_t = property_value<clock_2x_key, std::bool_constant<Enable>>;
+  };
+
+  enum class ram_stitching_enum : std::uint16_t { min_ram, max_fmax };
+
+  struct ram_stritching_key {
+    template <ram_stritching_enum Ram_stritching>
+    using value_t = property_value<
+        ram_stritching_key,
+        std::integral_constant<ram_stritching_enum, Ram_stritching>>;
+  };
+
+  struct private_copies_key {
+    template <size_t n>
+    using value_t =
+        property_value<private_copies_key, std::integral_constant<n>>;
+  };
+
+  struct num_replicates_key {
+    template <size_t n>
+    using value_t =
+        property_value<num_replicates_key, std::integral_constant<n>>;
+  };
+
+  template <resource_enum r> inline constexpr resource_key::value_t<r> resource;
+  inline constexpr resource_key::value_t<resource_enum::mlab> resource_mlab;
+  inline constexpr resource_key::value_t<resource_enum::block_ram>
+      resource_block_ram;
+
+  template <size_t e> inline constexpr num_banks_key::value_t<e> num_banks;
+
+  template <size_t e> inline constexpr stride_size_key::value_t<e> stride_size;
+
+  template <size_t e> inline constexpr word_size_key::value_t<e> word_size;
+
+  template <port_direction_enum d>
+  inline constexpr port_direction_key::value_t<d> port_direction;
+  inline constexpr port_direction_key::value_t<port_direction_enum::uni>
+      port_direction_uni;
+  inline constexpr port_direction_key::value_t<port_direction_enum::bi>
+      port_direction_bi;
+
+  template <bool b>
+  inline constexpr clock_2x_key::value_t<b> clock_2x;
+  inline constexpr clock_2x_key::value_t<true> clock_2x_true;
+  inline constexpr clock_2x_key::value_t<false> clock_2x_false;
+
+  template <ram_stitching_enum d>
+  inline constexpr ram_stitching_key::value_t<d> ram_stitching;
+  inline constexpr ram_stitching_key::value_t<ram_stitching_enum::min_ram>
+      ram_stitching_min_ram;
+  inline constexpr ram_stitching_key::value_t<ram_stitching_enum::max_fmax>
+      ram_stitching_max_fmax;
+
+  template <size_t n>
+  inline constexpr private_copies_key::value_t<n> private_copies;
+
+  template <size_t n>
+  inline constexpr num_replicates_key::value_t<n> num_replicates;
+
+} // namespace sycl::ext::intel::experimental
+----
+
+[frame="topbot",options="header"]
+|===
+|Property |Description
+
+a|
+[source,c++]
+----
+resource
+----
+a|
+Specifies which FPGA memory resources to use to implement the variable.
+
+The following values are supported:
+
+* `mlab`: data is stored in special Adaptive Logic Modules (ALMs) called memory-logic array blocks 
+* `block_ram`: data is stored in memory blocks, ie. M20Ks
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+num_banks
+----
+a|
+Number of banks that the array is divided into. If `stride_size` property is not specified in conjunction with `num_banks` property, memory will be divided using cyclic partitioning strategy, where consecutive words will be placed in different banks.
```

Comment:
We don't need to be so sparing with our words here:

```suggestion
The number of banks that the array is divided into. If the `stride_size` property is not specified in conjunction with the `num_banks` property, the memory will be divided using a cyclic partitioning strategy, where consecutive words will be placed in different banks.
```

Reply:
On second though, I'd rather we swap the second part of this description with the one for stride_size below.  I think this way of explaining num_banks and stride_size puts the relevant information in the wrong row of the table.  In the description of num_banks we should focus on how the number of banks is set.  

So instead I'd say:
The number of banks that the array is divided into.  If this is unspecified and stride_size is specified, a blocking partitioning strategy will be used, where consecutive words will be placed in the same bank until `stride_size` elements is reached, at which point further words will be added to a new bank.

You could even add:
This will result in elements/stride_size rounded up to the next power of 2 banks.

Reply:
I don't like this. Now the definition of how the property `num_banks` includes what happens when it is not specified, but not what happens when it is specified.